e2e/compatibility/consul_test.go

```diff
@@ -161,22 +163,30 @@ func testServiceInstanceCompatibility(t *testing.T, tempDir string, port int) {
 	testutils.CheckDir(t, false, webResourcesPath)
 
 	// 1. register db service instances
+	beforeRegister := time.Now()
```

Comment:
Just wondering if there is a big difference between just calling 
api.WaitForEvent(t, cts, "db_task", time.Now(), defaultWaitForEvent) rather than storing the time before doing the registration? Since we're not doing any validation on how long it takes, maybe it makes it a bit simpler?

Reply:
Not a big difference, it's all down to timing. That value is used by WaitForEvent for checking for an event that starts after the given time. To me, it's safer to have this now value set before the Consul registration on the off chance an event starts between the `registerService()` and the `api.WaitForEvent()`. But you could also argue that we don't want an event to happen between the `time.Now()` and when the service is finished registering in Consul!

Either way, I don't think we've ever had any issue with the placement of time.Now(), so I'm fine to change it as an inline variable to simplify things!

Reply:
I looked at the other tests and for the most part, we do the check beforehand, so I've left it as is!